src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100, hidden = false) {
+    this.name = name;
+    this.health = health;
+    this.hidden = hidden;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    const animalIndex = Animal.alive.indexOf(target);
+
+    if (target.hidden !== true && target instanceof Herbivore) {
+      target.health -= 50;
+    }
+
+    if (target.health <= 0) {
+      Animal.alive.splice(animalIndex, 1);
```

Comment:
The current implementation uses nested loops (indexOf + splice) which can be inefficient. Instead of finding the index and then splicing, you could use the `filter` method to remove the dead animal from the `Animal.alive` array in a more performant way.